doc/lstm.txt

```diff
@@ -1,14 +1,175 @@
 .. _lstm:
 
-LSTM Network for Sentiment Analysis
+LSTM Networks for Sentiment Analysis
 **********************************************
 
 Summary
 +++++++
 
-This tutorial aims to provide an example of how a Recurrent Neural Network (RNN) using the Long Short Term Memory (LSTM) architecture can be implemented using Theano. In this tutorial, this model is used to perform sentiment analysis on movie reviews from the `Large Movie Review Dataset <http://ai.stanford.edu/~amaas/data/sentiment/>`_, sometimes known as the IMDB dataset. 
+This tutorial aims to provide an example of how a Recurrent Neural Network
+(RNN) using the Long Short Term Memory (LSTM) architecture can be implemented
+using Theano. In this tutorial, this model is used to perform sentiment
+analysis on movie reviews from the `Large Movie Review Dataset
+<http://ai.stanford.edu/~amaas/data/sentiment/>`_, sometimes known as the
+IMDB dataset.
+
+In this task, given a movie review, the model attempts to predict whether it
+is positive or negative. This is a binary classification task.
+
+Data
+++++
+
+As previously mentionned, the provided scripts are used to train a LSTM
+recurrent neural on the Large Movie Review Dataset dataset.
+
+While the dataset is public, in this tutorial we provide a copy of the dataset
+that has previously been preprocessed according to the needs of this LSTM
+implementation. Running the code provided in this tutorial will automatically
+download the data to the local directory.
+
+Model
++++++
+
+LSTM
+====
+
+In a *traditional* recurrent neural network, during the gradient
+back-propagation phase, the gradient signal can end up being multiplied a
+large number of times (as many as the number of timesteps) by the weight
+matrix associated with the connections between the neurons of the recurrent
+hidden layer. This means that, the magnitude of weights in the transition
+matrix can have a strong impact on the learning process.
+
+If the weights in this matrix are small, it can lead to a situation called
```

Comment:
It's probably better to explicitly say that "the leading eigenvalue of the weight matrix is small (< 1)" or "large (> 1)".